src/components/Tags.astro

```diff
@@ -0,0 +1,41 @@
+---
+import {getCollection} from 'astro:content';
+
+interface Props {
+	tags?: string[];
+}
+const allResources = await getCollection('resources');
+
+// Set is used to remove duplicates
+const allTags = Array.from(
+	new Set(allResources.flatMap((resource) => resource.data.tags ?? []))
+);
+const {tags} = Astro.props;
+
+// Determine which tags to display
+const filteredTags =
+	tags && tags.length > 0
+		? allTags.filter((tag) => tags.includes(tag))
+		: allTags;
+---
+
+<style>
+	.tag {
+		display: inline-block;
+		background-color: #f0f0f0;
+		color: #333;
+		padding: 0.5em 1em;
+		margin: 0.5em;
+		border-radius: 0.25em;
+		font-size: 0.875em;
+		text-transform: capitalize;
+	}
+	.tags-container {
+		display: flex;
+		flex-wrap: wrap;
+	}
+</style>
+
+<div class="tags-container">
+	{filteredTags.map((tag) => <span class="tag">{tag}</span>)}
+</div>
```

Comment:
Could we add a list of all resources to this page, which can be filtered if someone clicks on one of the tags from the list?

Reply:
After looking at this myself, it is not that easy.. Need to investigate further, will update later.